### PR TITLE
Extract given entities values in JSON format

### PR DESCRIPTION
**Pull request by sinny777**
If your PR is related to a contribution to the taxonomy, please, fill
out the following questionnaire. If not, replace this whole text and the
following questionnaire with whatever information is applicable to your PR.

**Describe the contribution to the taxonomy**

- Extracting entities values in JSON format from given tables data
- Extracting entities values specially from financial reports
- If entity value which is asked to be extracted is not available then show null in JSON output
- Extract entities for each year in proper JSON format

**Input given at the prompt**

```
<|user|>
Generate a JSON output that extracts the entities given below surrounded by triple equals sign from given tables for each year. Return the output in JSON format array if exact match of entity name is found in given tables. If you do not find certain entity in the given tables then show its extracted entity value as "null". Make sure to extract entities values for each year. Do not provide any text or explanation along with JSON output.
===
["Total current assets", "Total current liabilities", "Total liabilities", "Total Revenue", "Total operating expenses", "Long-term mortgage debt"]
===

[Document]
Context from Page No - 97:
INDEX TO CONSOLIDATED FINANCIAL STATEMENTS > REPORT OF INDEPENDENT REGISTERED PUBLIC ACCOUNTING FIRM > Critical Audit Matter > Estimated Liability for Clinical Development Costs
 |  | December 31,
 | --- | ---
 |  | 2022 | 2021
 | --- | --- | ---
 | ASSETS
 | Current assets: Cash and cash equivalents | $ 276,472 | $ 869,191
 | Short-term investments | 1,710,397 | 1,245,782
 | Contracts receivable | 25,538 | 61,896
 | Inventories | 22,033 | 24,806
 | Other current assets | 168,254 | 143,374
 | Total current assets | 2,202,694 | 2,345,049
 | Property, plant and equipment, net | 74,294 | 178,069
 | Right-of-use assets. | 181,544 | 17,974
 | Deposits and other assets | 75,344 | 70,598
 | Total assets. | $ 2,533,876 | $ 2,611,690
 | LIABILITIES AND STOCKHOLDERS’ EQUITY
 | Current liabilities: Accounts payable. | $ 17,921 | $ 11,904
 | Accrued compensation | 49,178 | 38,810
 | Accrued liabilities | 140,101 | 88,560
 | Income taxes payable | 6,249 | 36
 | Current portion of deferred contract revenue. | 90,577 | 97,714
 | Other current liabilities | 7,535 | 3,526
 | Total current liabilities | 311,561 | 240,550
 | Long-term deferred contract revenue | 287,768 | 351,879
 | 0 percent convertible senior notes, net | 622,242 | 619,119
 | 0.125 percent convertible senior notes, net | 544,504 | 542,314
 | Long-term lease liabilities. | 178,941 | 19,432
 | Long-term mortgage debt | 8,847 | 59,713
 | Long-term obligations | 7,126 | 6,946
 | Total liabilities | 1,960,989 | 1,839,953
 | Stockholders’ equity: Common stock, $0.001 par value; 300,000,000 shares authorized, 142,057,736 and 141,210,015 shares issued and outstanding at December 31, 2022 and December 31, 2021, respectively | 142 | 141
 | Additional paid-in capital | 2,059,850 | 1,964,167
 | Accumulated other comprehensive loss | (57,480) | (32,668)
 | Accumulated deficit. | (1,429,625) | (1,159,903)
 | Total stockholders’ equity | 572,887 | 771,737
 | Total liabilities and stockholders’ equity | $ 2,533,876 | $ 2,611,690
[End]

<|assistant|>

```

**Response from the original model**

```
[
  {
    "Entity Name": "Total current assets",
    "Year": 2022,
    "Value": 2,202,694
  },
  {
    "Entity Name": "Total current liabilities",
    "Year": 2022,
    "Value": 311,561
  },
  {
    "Entity Name": "Total liabilities",
    "Year": 2021,
    "Value": 1,960,989
  },
  {
    "Entity Name": "Total Revenue",
    "Year": 2022,
    "Value": "Total revenue"
  },
  {
    "Entity Name": "Total operating expenses",
    "Year": 2022,
    "Value": "Total operating expenses"
  },
  {
    "Entity Name": "Long-term mortgage debt",
    "Year": 2022,
    "Value": 8,847
  }
]

```

**Response from the fine-tuned model**

```
{
    "Total current assets": {
        "2022": 276472,
        "2021": 869191
    },
    "Total current liabilities": {
        "2022": 311561,
        "2021": 240550
    },
    "Total liabilities": {
        "2022": 1960989,
        "2021": 1839953
    },
    "Total revenue": {
        "2022": null,
        "2021": null
    },
    "Total operating expenses": {
        "2022": null,
        "2021": null
    },
    "Long-term mortgage debt": {
        "2022": 8847,
        "2021": 59713
    }
}

```

**Contribution checklist**

- [X] The contribution was tested with `ilab generate`
- [X] No errors or warnings were produced by `ilab generate`
- [X] All [commits are signed off](https://github.com/instructlab/taxonomy/blob/main/CONTRIBUTING.md#legal) (DCO)
- [X] The `qna.yaml` file contains at least 5 `seed_examples`
- [X] The `qna.yaml` file was [linted](https://yamllint.com) and [prettified](https://onlineyamltools.com/prettify-yaml) ([yaml-validator](https://jsonformatter.org/yaml-validator) can do both)
- [X] An `attribution.txt` file in the same folder as the `qna.yaml` file
- [X] Content does not include PII or otherwise sensitive or confidential information
- [X] Content does not include anything documented in the project's [Avoid these Topics](https://github.com/instructlab/taxonomy/blob/main/docs/SKILLS_GUIDE.md#avoid-these-topics) guidelines

### CODE REVIEW

**Comment by jjasghar** [comment]
@instructlab-bot precheck

**Comment by jjasghar** [comment]
With the 0.18 release we are rebuilding our backends for the `precheck` commands and verifications. Thank you for this, but we can't verify it until that is built up.

File: compositional_skills/extraction/annual_report/json/qna.yaml [hunk 1]
```diff
@@ -0,0 +1,596 @@
+version: 2
```
[the remark this answers is not in the document]
Comment (sinny777) [reply compositional_skills/extraction/annual_report/json/qna.yaml]: Done

**Comment by sinny777** [comment]
I hope there's no change pending from my end and this skill will be added to the model in future release.

Thanks,
Gurvinder Singh

**Comment by mcorbin-ibm** [comment]
@jjasghar I reviewed this from a taxonomy tree perspective, and I have these questions:

1. Is a financial report a specific type of annual report?  The qna is all about financial reports data, not an annual report?
2. Since this is a compositional skill, does it really matter that it is for an annual report or financial report?  

If it is important to keep the context as part of this categorization, then I would put it at the end -- something like this:

compositional_skills/grounded/technology/data_extraction/json/financial_reports

**Comment by jjasghar** [comment]
@instructlab-bot precheck

**Comment by vishnoianil** [comment]
@instructlab-bot precheck

**Comment by jjasghar** [comment]
After some discussion with the triage team, being this is _very_ close to the code concept of creating a `json` blob from data, this falls under the same reasoning that https://github.com/instructlab/taxonomy/pull/1058 has. If our model was a "code" model, this may have a reasonable chance of creating this, but being our `granite-7b-lab` model _isn't_ a code model, we are going to reject this PR.

Thank you for taking the time, but this is a bad fit for this model at this time.

**Pull request closed without merging**